examples/main.cpp

```diff
@@ -46,37 +46,14 @@
 //@HEADER
 */
 
-#ifndef PRESSIO_CMAKE_CONFIG_H_
-#define PRESSIO_CMAKE_CONFIG_H_
+#include <pressio/utils.hpp>
 
-#cmakedefine PRESSIO_DEBUG_BUILD
-#cmakedefine PRESSIO_ENABLE_DEBUG_PRINT
-#cmakedefine PRESSIO_ENABLE_TPL_EIGEN
-#cmakedefine PRESSIO_ENABLE_TPL_TRILINOS
-#cmakedefine PRESSIO_ENABLE_TEUCHOS_TIMERS
-#cmakedefine PRESSIO_ENABLE_TPL_PYBIND11
-#cmakedefine PRESSIO_ENABLE_TPL_KOKKOS
-#cmakedefine PRESSIO_ENABLE_TPL_BLAS
-#cmakedefine PRESSIO_ENABLE_TPL_LAPACK
-#cmakedefine PRESSIO_ENABLE_TPL_MPI
-#cmakedefine PRESSIO_ENABLE_CXX14
-#cmakedefine PRESSIO_ENABLE_CXX17
-#cmakedefine PRESSIO_ENABLE_CXX20
+int main() {
+  pressio::log::initialize(pressio::logto::terminal, "log.txt");
+  pressio::log::setVerbosity({pressio::log::level::info});
 
-// macros for catching compiler
-#if defined( __INTEL_COMPILER )
-#define PRESSIO_COMPILER_INTEL __INTEL_COMPILER
-#endif
+  double a = 1.;
+  PRESSIOLOG_INFO("pressio log message, {:3}", a);
 
-#if !defined( __clang__ ) && !defined( PRESSIO_COMPILER_INTEL ) &&defined( __GNUC__ )
-#define PRESSIO_COMPILER_GNU __GNUC__*100+__GNUC_MINOR__*10+__GNUC_PATCHLEVEL__
-#endif
-
-// macros for deprecation (need to adjust when we can move to c++14)
-#if defined(PRESSIO_COMPILER_GNU) || defined(PRESSIO_COMPILER_CLANG)
-#define PRESSIO_DEPRECATED __attribute__ ((deprecated))
-#else
-#define PRESSIO_DEPRECATED
-#endif
-
-#endif
+  pressio::log::finalize();
+}
```

Comment:
and also, if i am not mistken, i think the following would still work:

```
#define PRESSIO_ENABLE_TPL_EIGEN
#define PRESSIO_LOG_LEVEL_INFO

#include <pressio/utils.hpp>

int main() {
  pressio::log::initialize(pressio::logto::terminal, "log.txt");
  pressio::log::setVerbosity({pressio::log::level::info});
  double a = 1.;
  PRESSIOLOG_INFO("pressio log message, {:3}", a);
  pressio::log::finalize();
}
```

Reply:
This works fine and is added to the CI now.

Reply:
Note that we cannot use a simple `static_assert` when using `#define` like this:
https://github.com/Pressio/pressio/blob/0dc3d12e2eab8fd31dd3fc07b0113f35c166966a/include/pressio/macros.hpp#L120

```
./include/pressio/macros.hpp:120:39: error: expected expression
static_assert(PRESSIO_ENABLE_TPL_EIGEN, "Eigen is not enabled.");
```